operators/multiclusterobservability/controllers/placementrule/hub_info_secret.go

```diff
@@ -69,9 +69,13 @@ func generateHubInfoSecret(client client.Client, obsNamespace string,
 		}
 	}
 
-	obsApiURL := url.URL{
-		Host: obsAPIHost,
-		Path: operatorconfig.ObservatoriumAPIRemoteWritePath,
+	if !strings.HasSuffix(obsAPIHost, operatorconfig.ObservatoriumAPIRemoteWritePath) {
+		obsAPIHost += operatorconfig.ObservatoriumAPIRemoteWritePath
```

Comment:
We could use path.Join in case there is trailing slash to handle in obsAPIHost?

Reply:
I decided to go with `net/url.JoinPath` (which is also present in `url.URL.JoinPath`) instead. `path.Join` is made for filesystem paths and has filesystem specific behaviors that are not a good idea for URL path joining.